[python_legacy] BOTO_STS_CLIENT lazy initialization

puchengy opened the pull request:
Importing s3_filesystem module will initialize boto client which will generate network IO, thus make this as lazy initialization.

I can't find similar thing in python module right now, please suggest if I should change other places for similar reasons.

Fokko reviewed (approved):
If I would write this, I would create a function for it:
```python
@cache
def get_sts_client()
  return boto3.client('sts')
```
But since this is the legacy one, I'm fine with this. Keep in mind that we're not releasing this code.

Would be awesome if we can also port this functionality to the new Python library.

puchengy commented:
@Fokko I addressed the comment, please take another look. 

> port this functionality to the new Python library.

Also what do you mean by this? As I can't see similar ways of constructing boto3 client for sts client.
